### PR TITLE
Add upload priority layers

### PR DESCRIPTION
**Pull request by zamuzakki**
This is PR for https://github.com/ConservationInternational/cplus-plugin/issues/441

I have tested using several PWL, and give them some weight.
image

The API then considers those weight, as indicated in this log

```
[2024-05-29 07:14:39,165: INFO/ForkPoolWorker-3]  Used parameters for calculating weighting activities {'CELLSIZE': 0, 'CRS': None, 'EXPRESSION': '"Alien_Plant_Removal_3aa5@1" + (5.0*"social_int_clip_norm@1") + (2.0*"ei_all_gknp_clip_norm@1") + (5.0*"biocombine_clip_norm_QtBE9kG@1") + (3.0*"Policy_S2nhJ2z@1")', 'EXTENT': '877301.8070349816,1016212.7812331037,7203080.7869933,7345132.766826201 [EPSG:32735]', 'LAYERS': ['/home/web/media/4/77405c03-df4e-4b41-8701-66a21239c3c1/normalized_activities/Alien_Plant_Removal_3aa5.tif', '/home/web/media/4/77405c03-df4e-4b41-8701-66a21239c3c1/priority_layer/social_int_clip_norm.tif', '/home/web/media/4/77405c03-df4e-4b41-8701-66a21239c3c1/priority_layer/ei_all_gknp_clip_norm.tif', '/home/web/media/4/77405c03-df4e-4b41-8701-66a21239c3c1/priority_layer/biocombine_clip_norm_QtBE9kG.tif', '/home/web/media/4/77405c03-df4e-4b41-8701-66a21239c3c1/priority_layer/Policy_S2nhJ2z.tif'], 'OUTPUT': '/home/web/media/4/77405c03-df4e-4b41-8701-66a21239c3c1/weighted_activities/Alien_Plant_Removal_81ef.tif'}
```

I have also set Coefficient for carbon layers in settings. That value and layer is reflected in this log
```
[2024-05-29 07:14:27,953: INFO/ForkPoolWorker-3] Used parameters for combining pathways and carbon layers generation: {'CELLSIZE': 0, 'CRS': None, 'EXPRESSION': '"Final_Avoided_Grassland_priority_norm_new@1" + 1.0 * ("bou_SOC_carbonsum_norm_null_con_clip@1")', 'EXTENT': '877301.8070349816,1016212.7812331037,7203080.7869933,7345132.766826201 [EPSG:32735]', 'LAYERS': ['/home/web/media/4/77405c03-df4e-4b41-8701-66a21239c3c1/ncs_pathway/Final_Avoided_Grassland_priority_norm_new.tif', '/home/web/media/4/77405c03-df4e-4b41-8701-66a21239c3c1/ncs_carbon/bou_SOC_carbonsum_norm_null_con_clip.tif'], 'OUTPUT': 'TEMPORARY_OUTPUT'} 
```

### CODE REVIEW

File: src/cplus_plugin/api/scenario_task_api_client.py [hunk 1]
```diff
@@ -258,6 +249,14 @@ def upload_layers(self):
             Settings.SIEVE_ENABLED, default=False, setting_type=bool
         )
 
+        priority_layers = settings_manager.get_priority_layers()
```
Comment (danangmassandy) [inline src/cplus_plugin/api/scenario_task_api_client.py]: please change to self.get_priority_layers()

File: src/cplus_plugin/api/scenario_task_api_client.py [hunk 2]
```diff
@@ -258,6 +249,14 @@ def upload_layers(self):
...
+        for priority_layer in priority_layers:
+            if priority_layer['uuid'] in activity_pwl_uuids and os.path.exists(priority_layer['path']):
```
Comment (danangmassandy) [inline src/cplus_plugin/api/scenario_task_api_client.py]: I think it's better to use priority_layer.get so we don't get key error exception. The original code in tasks.py is also using .get('").
Please apply the same when retrieving 'groups' and 'value'.

File: src/cplus_plugin/api/scenario_task_api_client.py [hunk 3]
```diff
@@ -469,19 +468,16 @@ def build_scenario_detail_json(self):
...
-                                priority_layer["path"]
-                            )["uuid"]
-                            priority_layer["layer_uuid"] = priority_layer["uuid"]
-                            new_priority_layers.append(priority_layer)
-            activity["priority_layers"] = new_priority_layers
+            activity["priority_layers"] = [
+                p for p in activity["priority_layers"] if p
+            ]
+
+        priority_layers = settings_manager.get_priority_layers()
```
Comment (danangmassandy) [inline src/cplus_plugin/api/scenario_task_api_client.py]: please change to self.get_priority_layers()

File: src/cplus_plugin/api/scenario_task_api_client.py [hunk 4]
```diff
@@ -469,19 +468,16 @@ def build_scenario_detail_json(self):
...
+        for priority_layer in priority_layers:
+            if priority_layer['path'] in self.path_to_layer_mapping:
```
Comment (danangmassandy) [inline src/cplus_plugin/api/scenario_task_api_client.py]: same as above comment, use priority_layer.get("")

**Review by danangmassandy — CHANGES_REQUESTED** [review]

**Comment by danangmassandy** [comment]
@zamuzakki please help to resolve the conflict and then I will merge this PR! Thank you,

**Comment by zamuzakki** [comment]
@danangmassandy I've resolved the conflict

**Review by danangmassandy — APPROVED** [review]
Cool, thanks Zakki

**Pull request merged**